providers/asrockrack/firmware.go

```diff
@@ -182,6 +182,13 @@ func (a *ASRockRack) firmwareUpdateStatus(ctx context.Context, component string,
 		return "", errors.Wrap(bmclibErrs.ErrFirmwareInstallStatus, err.Error())
 	}
 
+	if progress == nil && installStatus == versionStrMatch {
+		// TODO: we should pass the force parameter to firmwareUpdateStatus,
+		// so that we can know if we expect a version change or not
+		a.log.V(3).Info("Nil progress + no version change -> unknown")
+		return constants.FirmwareInstallUnknown, nil
+	}
+
 	switch installStatus {
 	case versionStrMatch:
```

Comment:
I think you can move the added conditional `if progress == nil { return constants.FirmwareInstallUnknown} }` under this switch case?

Reply:
Sure.